pkg/core/blockchain.go

```diff
@@ -971,6 +972,8 @@ func (bc *Blockchain) VerifyWitnesses(t *transaction.Transaction) error {
 	if len(hashes) != len(witnesses) {
 		return errors.Errorf("expected len(hashes) == len(witnesses). got: %d != %d", len(hashes), len(witnesses))
 	}
+	sort.Slice(hashes, func(i, j int) bool { return hashes[i].Less(hashes[j]) })
```

Comment:
I would probably add a godoc to VerifyWitnesses func about ordering in hashes and winesses. So it's not a surprise =)